### ISSUE TITLE
[NPC][WotlK] Ice Steppe Bull, Ice Steppe Rhino Part 2 - missing spawns/paths

### ISSUE THREAD

**Issue by Teridan-dal-wow** [dalaranwow/dalaran-wow#4403]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Npc Ice Steppe Bull http://wotlk.openwow.com/npc=30445 , Ice Steppe Rhino http://wotlk.openwow.com/npc=29469 have few missing paths and spawn locations. Area relevent to this report:
![image](https://cloud.githubusercontent.com/assets/6797611/22853239/a9e5b1be-f059-11e6-8703-d8cac9ae686c.png)

**Expected behaviour**: 
 - Part 1 - Suggestion: increase the spawndist of creature (rhino without paths):
`DB GUID: 98751 --- to 10 ` shown below to match retail.
![image](https://cloud.githubusercontent.com/assets/6797611/22853237/8dd3376c-f059-11e6-9868-617f7f263619.png)
Small video sample from retail: https://www.youtube.com/watch?v=j5eDFoyNn38&feature=youtu.be

Update: After checking all of the rhinos that do not have paths, i noticed they all share the same random movement radius as the one pointed above. Here is the list of the ones that are present on the server:

> DB GUID: 98751  ---This one might be wrong, i tweak with him.
> DB GUID: 98750
> DB GUID: 98790
> DB GUID: 98789 ---- This one should have DisplayID: 26285 
> DB GUID: 98795
> DB GUID: 98791
> DB GUID: 98794
> DB GUID: 98792

 - Part 2 - Change the model of creature
`DB GUID: 98760, DisplayID: 27006 `
to
`DB GUID: 98760, DisplayID: 26285 `
Screenshot from the server:
![wowscrnshot_021117_131019](https://cloud.githubusercontent.com/assets/6797611/22853327/2fecdfa6-f05c-11e6-9110-f6e7d383a6e3.jpg)
Screenshot from retail:
![image](https://cloud.githubusercontent.com/assets/6797611/22853340/87d857e0-f05c-11e6-9e3a-41c5efa7b5bf.png)

 - Part 3 - There is an issue with the groups of rhinos that patrol as packs, once they get to a wp all 3 of them head to the next one in a straight line, instead only the pack leader should do that and the rest should follow him (if that is not possible to be done a workaround can be to add more wps in their paths so this does not happen).
Example for the server:
![wowscrnshot_021117_131033](https://cloud.githubusercontent.com/assets/6797611/22853378/980ac962-f05d-11e6-9ded-74c6ce3113ec.jpg)
Small video sample from retail: 
https://www.youtube.com/watch?v=CVjty2hV5tA&feature=youtu.be
Note: The 2 rhinos that follow the bull are basicly "mirroring" his steps.

 - Part 4 - Missing and incorrect spawns:
This spawn location of this one needs to be moved slightly on the left:
![wowscrnshot_021117_134143](https://cloud.githubusercontent.com/assets/6797611/22853497/7bd911c4-f060-11e6-8a4d-12d02561921b.jpg)
Missing spawns:
![wowscrnshot_021117_134158](https://cloud.githubusercontent.com/assets/6797611/22853514/e25ce3b2-f060-11e6-9e87-c60c197f6c94.jpg)
Wrong spawns:
![wowscrnshot_021117_134212](https://cloud.githubusercontent.com/assets/6797611/22853538/96025802-f061-11e6-962b-70c719405872.jpg)

Video from retail legion from 11.02.2017:
https://www.youtube.com/watch?v=QAiRgSwj26I&feature=youtu.be
Note: Check sniff part 1

 - Part 5 - Missing paths:
These 3 rhinos are missing paths:
![image](https://cloud.githubusercontent.com/assets/6797611/22853677/8b12a9ac-f065-11e6-81d7-ad1247dfdde8.png)
Also 1 of them should have a "big" model (DisplayID: 26285) and there should be 1 more big one (that i coundn't find on the server).
Video from retail legion from 11.02.2017:
https://www.youtube.com/watch?v=U98gEM5Gx2M&feature=youtu.be
Note: Check sniff part 2

 - Part 6 - Few more missing spawns (on the field under the dragons):
These are all small ones (DisplayID: 27006):
![wowscrnshot_021117_142956](https://cloud.githubusercontent.com/assets/6797611/22853753/05b9563c-f067-11e6-96a0-9c9d876b8248.jpg)
![wowscrnshot_021117_142841](https://cloud.githubusercontent.com/assets/6797611/22853752/05b862c2-f067-11e6-8c0c-3c88722e6001.jpg)
![wowscrnshot_021117_140113](https://cloud.githubusercontent.com/assets/6797611/22853754/05c8ecc8-f067-11e6-9404-275bc4fc54fb.jpg)
![wowscrnshot_021117_143123](https://cloud.githubusercontent.com/assets/6797611/22853755/05ce12a2-f067-11e6-8e37-4a755a33a2e4.jpg)
And 1 big one (DisplayID: 26285):
![wowscrnshot_021117_143033](https://cloud.githubusercontent.com/assets/6797611/22853764/30a6c776-f067-11e6-9c38-bfb945b201b6.jpg)
Note: Sniffing these is very hard as they constantly die to drakes and i have to wait for respawns = massive sniff file.

**Comment by Rushor** [comment]
concerning Part 3 - does this only affect that group shown in the picture or all groups?

**Comment by Teridan-dal-wow** [comment]
It affects all groups in that area, also i would like to add that some of the formations looks abit strange (maybe due to this issue), here are few examples:
![wowscrnshot_021117_154819](https://cloud.githubusercontent.com/assets/6797611/22854198/ca472498-f071-11e6-8fce-cd309737069a.jpg)
![wowscrnshot_021117_154828](https://cloud.githubusercontent.com/assets/6797611/22854199/ca49e336-f071-11e6-8695-d46361cc3a79.jpg)

**Comment by Teridan-dal-wow** [comment]
Part 1,2,4,5,6 of this report revisited and corrected.
Part 3 of this report ignored (will post new issue for it with more information).
What is done here:
 - removed few wrong spawns
 - corrected few models 
 - corrected spawn loc of few creatures
 - added 2 formations
 - spawned all missing rhinos on the field
 - added 1 path
How it looks like:
![image](https://cloud.githubusercontent.com/assets/6797611/23102294/6d256dbc-f6af-11e6-82e6-56787613d2ec.png)
![image](https://cloud.githubusercontent.com/assets/6797611/23102298/77a3e516-f6af-11e6-8112-0aa2c46a2775.png)
![image](https://cloud.githubusercontent.com/assets/6797611/23102301/7c5cf476-f6af-11e6-9a4a-88edddda2cfb.png)

``` sql
UPDATE `creature` SET `modelid` = 26285 WHERE `guid` IN (98760, 98789, 98759, 98791);
UPDATE `creature` SET `position_x` = 7285.172, `position_y` = -990.1328 , `position_z` = 915.6776, `spawndist` = 10 WHERE `guid` = 98751; 
UPDATE `creature` SET `spawndist` = 10 WHERE `guid` IN (98750, 98790, 98789, 98795, 98791, 98794, 98792);
UPDATE `creature` SET `position_x` = 7260.41, `position_y` = -1475.84 , `position_z` = 916.793 WHERE `guid` IN (98747, 98746, 98734);

 
SET @cguid:= 5004976;
DELETE FROM `creature` WHERE `guid` IN (98733, 98764);
DELETE FROM `creature` WHERE `guid` BETWEEN @CGUID+0 AND @cguid+9;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+0, 29469, 571, 0, 0, 1, 1, 26285, 0, 7444.167, -850.168, 915.8319, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+1, 29469, 571, 0, 0, 1, 1, 26285, 0, 7354.931, -917.179, 909.835, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+2, 29469, 571, 0, 0, 1, 1, 0, 0, 7586.971, -1257.549, 925.662, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+3, 29469, 571, 0, 0, 1, 1, 27006, 0, 7586.971, -1257.549, 925.662, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+4, 29469, 571, 0, 0, 1, 1, 26285, 0, 7352.417, -1489.691, 927.1118, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+5, 29469, 571, 0, 0, 1, 1, 26285, 0, 7352.417, -1489.691, 927.1118, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+6, 29469, 571, 0, 0, 1, 1, 26285, 0, 7352.417, -1489.691, 927.1118, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+7, 29469, 571, 0, 0, 1, 1, 0, 0, 7180.141, -1479.33, 923.094, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+8, 29469, 571, 0, 0, 1, 1, 0, 0, 7313.333, -1484.522, 920.257, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420), -- 29469 (Area: -1) (possible waypoints or random movement)
(@CGUID+9, 29469, 571, 0, 0, 1, 1, 0, 0, 7269.304, -1386.076, 909.406, 2.979512, 300, 10, 0, 1, 0, 1, 0, 0, 0, 23420); -- 29469 (Area: -1) (possible waypoints or random movement)

DELETE FROM `creature_addon` WHERE `guid` IN (98733, 98764);
DELETE FROM `creature_addon` WHERE `guid` BETWEEN @CGUID+0 AND @cguid+9;
INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `auras`) VALUES
(@CGUID+0, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+1, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+2, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+3, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+4, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+5, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+6, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+7, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+8, 0, 0, 0, 1, 0, ''), -- 29469
(@CGUID+9, 0, 0, 0, 1, 0, ''); -- 29469


DELETE FROM `creature_formations` WHERE `leaderguid` = 98793;
INSERT INTO `creature_formations` (`leaderGUID`, `memberGUID`, `dist`, `angle`, `groupAI`, `point_1`, `point_2`) VALUES
(98793,98793,0,0,2,0,0),
(98793,98746,20,0,2,0,0),
(98793,98747,9,330,2,0,0),
(98793,98734,9,30,2,0,0);

-- Pathing for Ice Steppe Rhino Entry: 29469 'TDB FORMAT' 
SET @NPC := @CGUID+3;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=7352.417,`position_y`=-1489.691,`position_z`=927.1118 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,7352.417,-1489.691,927.1118,0,0,0,0,100,0), -- 14:06:51
(@PATH,2,7388.675,-1470.793,929.7119,0,0,0,0,100,0), -- 14:07:05
(@PATH,3,7410.477,-1451.502,932.8225,0,0,0,0,100,0), -- 14:07:22
(@PATH,4,7438.898,-1428.052,934.5678,0,0,0,0,100,0), -- 14:07:34
(@PATH,5,7470.145,-1403.208,934.2198,0,0,0,0,100,0), -- 14:07:53
(@PATH,6,7495.443,-1352.971,930.0919,0,0,0,0,100,0), -- 14:08:06
(@PATH,7,7507.908,-1315.325,923.4832,0,0,0,0,100,0), -- 14:08:27
(@PATH,8,7506.533,-1277.36,916.8058,0,0,0,0,100,0), -- 14:08:44
(@PATH,9,7481.845,-1237.823,908.0139,0,0,0,0,100,0), -- 14:09:02
(@PATH,10,7466.036,-1237.008,906.6911,0,0,0,0,100,0), -- 14:09:18
(@PATH,11,7411.459,-1234.95,903.5399,0,0,0,0,100,0), -- 14:09:32
(@PATH,12,7368.853,-1239.925,902.4027,0,0,0,0,100,0), -- 14:09:48
(@PATH,13,7335.434,-1254.402,903.4313,0,0,0,0,100,0), -- 14:10:04
(@PATH,14,7299.958,-1282.759,905.9913,0,0,0,0,100,0), -- 14:10:18
(@PATH,15,7275.941,-1315.047,911.145,0,0,0,0,100,0), -- 14:10:36
(@PATH,16,7266.458,-1325.19,912.4014,0,0,0,0,100,0), -- 14:10:52
(@PATH,17,7232.259,-1370.342,916.2829,0,0,0,0,100,0), -- 14:11:06
(@PATH,18,7215.428,-1395.22,915.1907,0,0,0,0,100,0), -- 14:11:20
(@PATH,19,7199.2,-1427.229,916.132,0,0,0,0,100,0), -- 14:11:35
(@PATH,20,7202.649,-1454.023,916.8901,0,0,0,0,100,0), -- 14:11:47
(@PATH,21,7218.063,-1481.952,917.3933,0,0,0,0,100,0), -- 14:12:00
(@PATH,22,7247.573,-1517.014,923.032,0,0,0,0,100,0), -- 14:12:17
(@PATH,23,7254.085,-1518.963,923.8891,0,0,0,0,100,0), -- 14:12:29
(@PATH,24,7278.132,-1518.355,923.5187,0,0,0,0,100,0), -- 14:12:36
(@PATH,25,7320.935,-1502.203,923.3114,0,0,0,0,100,0); -- 14:12:49

DELETE FROM `creature_formations` WHERE `leaderguid` = @CGUID+3;
INSERT INTO `creature_formations` (`leaderGUID`, `memberGUID`, `dist`, `angle`, `groupAI`, `point_1`, `point_2`) VALUES
(@CGUID+3,@CGUID+3,0,0,2,0,0),
(@CGUID+3,@CGUID+4,20,0,2,0,0),
(@CGUID+3,@CGUID+5,9,330,2,0,0),
(@CGUID+3,@CGUID+6,9,30,2,0,0);
```
Also unrelated to this report, but as i spot i might aswell ask here. Npc http://wotlk.openwow.com/npc=30383 and npc http://wotlk.openwow.com/npc=30382 have `phasemask =1` which makes they visible to players without completing the chain and been phased.
On server:
![image](https://cloud.githubusercontent.com/assets/6797611/23102407/472f9c0c-f6b1-11e6-94a5-a36006852cc6.png)
On retail:
![image](https://cloud.githubusercontent.com/assets/6797611/23102415/7d5f3940-f6b1-11e6-9b7c-93fad1a40c82.png)

Suggestion?:
``` sql
UPDATE `creature` SET `phaseMask` = 2 WHERE `guid` IN (144231, 144232); 
```
But i am not really sure because if i abondone the quest (http://wotlk.openwow.com/quest=13417) this happens:
![image](https://cloud.githubusercontent.com/assets/6797611/23102453/1cec17d0-f6b2-11e6-9ce7-428420184fc5.png)

**Comment by Rushor** [comment]
> But i am not really sure because if i abondone the quest (http://wotlk.openwow.com/quest=13417) this 

can be fixed with spell_area. i will fix that for ya

**Comment by Rushor** [comment]
DB/Creature: Ice Steppe Bull, Ice Steppe Rhino Part 2 - Add missing spawns/paths + Apply phasing for Brann's Flying Machine + Brann

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4403
* thanks to Teridan
* commit: 59ad1338f6aef6311aabcc274989c1f802f36a69

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.
